src/op.rs

```diff
@@ -1566,17 +1566,39 @@ impl<'context, 'input, Endian> Evaluation<'context, 'input, Endian>
             Operation::Shl => {
                 let v1 = try!(self.pop());
                 let v2 = try!(self.pop());
-                self.push(v2 << v1);
+                // Because wrapping_shl takes a u32, not a u64, we do
+                // the check by hand.
+                if v1 >= 32 {
```

Comment:
Oh duh, it should be 64.  I'll add a test for that.
It doesn't have to match the target's type here, just whatever native type we're using in the implementation.